pixeldata/src/attribute.rs

```diff
@@ -76,6 +78,14 @@ pub enum GetAttributeError {
         value: String,
         backtrace: Backtrace,
     },
+
+    #[snafu(visibility(pub))]
+    #[snafu(display("Lengths must all be the same for `{:?}`, found `{:?}`", items, values))]
+    LengthMismatch {
```

Comment:
This enum variant tripped the error type size check. Large errors lead to large `Result`s, negatively affecting performance even in the happy path.

I think we can further decompose it into more specific variants with equally more specific details. It would go like this:

```rust
    /// value multiplicity of VOI LUT Function must match the number of frames ({voi_lut_function_vm} vs {number_of_frames})
    LengthMismatchVoiLutFunction {
        voi_lut_function_vm: u32,
        number_of_frames: u32,
        backtrace: Backtrace,
    },
    /// value multiplicity of Rescale Intercept and Rescale Slope must match the number of frames ({rescale_intercept_vm} & {rescale_slope_vm} vs {number_of_frames})
    LengthMismatchRescale {
        rescale_intercept_vm: u32,
        rescale_slope_vm: u32,
        number_of_frames: usize,
        backtrace: Backtrace,
    },
    /// value multiplicity of Window Center and Window Width must match the number of frames ({window_center_vm} & {window_width_vm} vs {number_of_frames})
    LengthMismatchWindowLevel {
        rescale_intercept_vm: u32,
        rescale_slope_vm: u32,
        number_of_frames: u32,
        backtrace: Backtrace,
    },
```

I believe this would constrain the size more nicely while conveying the same information.

Reply:
No problem, I will go ahead and do that, hope you had a good holiday season!

Reply:
Okay, updated the code, I wasn't familiar with how docstrings get formatted in snafu and the rest of the enum doesn't use docstrings, so I went with the specific `snafu(display())`, lmk if you want me to change that